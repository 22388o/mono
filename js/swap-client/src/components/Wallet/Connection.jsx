import React, { useCallback, useEffect, useState, useSyncExternalStore } from 'react';
import RectangleRoundedIcon from '@mui/icons-material/RectangleRounded';
//import { activitiesStore } from '../../syncstore/activitiesstore.js';
import { userStore } from '../../syncstore/userstore.js';
import { walletStore } from '../../syncstore/walletstore.js';
import { IndexedDB } from '@portaldefi/sdk';

export const ConnectionComponent = () => {
  const user = useSyncExternalStore(userStore.subscribe, () => userStore.currentState);
  const activities = useSyncExternalStore(IndexedDB.subscribe, IndexedDB.getAllActivities);
  const [swapActivities, setSwapActivities] = useState(activities);

  const logOut = useCallback(() => {
    // dispatch(signOut());
    walletStore.dispatch({ type: 'CLEAR_NODE_DATA' })
    walletStore.dispatch({ type: 'CLEAR_WALLET_DATA' })
    // return Promise.all([alice.stop(), bob.stop()]);
    return //Promise.all([user.user.stop()])
  }, [walletStore])

  useEffect(() => {
    setSwapActivities(activities);
  }, [activities])


  return (
    <h4 className='flex-center lightgreen'><RectangleRoundedIcon />&nbsp;All systems ok!</h4>
  );
}
