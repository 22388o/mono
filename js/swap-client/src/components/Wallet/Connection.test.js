import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom'; // For the "toBeInTheDocument" matcher
import { ConnectionComponent } from './Connection';

describe('Connection component test', () => {
  it('renders component', async () => {

    const {container} = render(
      <ConnectionComponent />
    );

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
