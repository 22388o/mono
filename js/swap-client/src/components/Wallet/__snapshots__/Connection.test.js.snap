// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection component test renders component 1`] = `
<div>
  <h4
    class="flex-center lightgreen"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="RectangleRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M2 6v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2z"
      />
    </svg>
    Â All systems ok!
  </h4>
</div>
`;
